File: remote/src/main/java/com/kelsos/mbrc/ui/dialogs/WebViewDialog.java
package com.kelsos.mbrc.ui.dialogs;

import android.app.Dialog;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.DialogFragment;
import android.webkit.WebView;
import com.afollestad.materialdialogs.MaterialDialog;

public class WebViewDialog extends DialogFragment {

  public final static String ARG_URL = "url";
  public final static String ARG_TITLE = "title";

  @NonNull @Override public Dialog onCreateDialog(Bundle savedInstanceState) {
    MaterialDialog.Builder builder = new MaterialDialog.Builder(getActivity());
    final WebView webView = new WebView(getActivity());
    webView.loadUrl(getArguments().getString(ARG_URL));
    builder.customView(webView, false);
    builder.title(getArguments().getInt(ARG_TITLE));
    builder.positiveText(android.R.string.ok);
    builder.onPositive((dialog, which) -> dialog.dismiss());
    return builder.build();
  }
}

File: app/src/main/java/com/kelsos/mbrc/ui/dialogs/WebViewDialog.kt
package com.kelsos.mbrc.ui.dialogs

import android.app.Dialog
import android.os.Bundle
import android.webkit.WebView
import androidx.fragment.app.DialogFragment
import com.google.android.material.dialog.MaterialAlertDialogBuilder

class WebViewDialog : DialogFragment() {
  override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
    val title = arguments?.getInt(ARG_TITLE) ?: throw Exception("argument null")
    val url = arguments?.getString(ARG_URL) ?: throw Exception("argument null")

    return MaterialAlertDialogBuilder(requireActivity())
      .setTitle(title)
      .setView(
        WebView(requireActivity()).apply {
          loadUrl(url)
        },
      ).setPositiveButton(android.R.string.ok) { dialog, _ -> dialog.dismiss() }
      .show()
  }

  companion object {
    const val ARG_URL = "url"
    const val ARG_TITLE = "title"
  }
}
